index.js

```diff
@@ -242,6 +252,49 @@ TAP_ELEMENT_SCRIPT = `function tapElement(params) {
   }
 }`;
 
+GET_ELEMENT_RECT_SCRIPT = `function getElementRect(params) {
+  var element = window._fridaElementByRef[params.ref];
+  var rect = element.getBoundingClientRect();
+  return {
+    rect: [[rect.left, rect.top], [rect.width, rect.height]]
+  };
+}`;
+
+/*
+ * Adapted from: http://stackoverflow.com/a/15203639/5418401
+ */
+IS_ELEMENT_VISIBLE_SCRIPT = `function isElementVisible(params) {
+  var element = window._fridaElementByRef[params.ref];
+  var rect = element.getBoundingClientRect();
+  if (rect.width == 0 || rect.height == 0) {
+    return {
+      visible: false
+    };
+  }
+
+  var vWidth = window.innerWidth || doc.documentElement.clientWidth;
+  var vHeight = window.innerHeight || doc.documentElement.clientHeight;
+  if (rect.right < 0 || rect.bottom < 0 || rect.left > vWidth || rect.top > vHeight) {
+    return {
+      visible: false
+    };
+  }
+
+  var efp = function(x, y) {
+    return document.elementFromPoint(x, y)
+  };
+
+  // Return true if any of its four corners are visible
+  return {
+    visible: (
+      element.contains(efp(rect.left, rect.top)) ||
+      element.contains(efp(rect.right, rect.top)) ||
+      element.contains(efp(rect.right, rect.bottom)) ||
+      element.contains(efp(rect.left, rect.bottom))
+    )
+  };
+}`;
```

Comment:
yes, i'll go with unboxing because `perform()` treats the return value as an object